File: AtividadePython/cadastro_clientes.py
class CadastroClientes:
    def __init__(self):
        self.cliente_list = []

    def adicionar_cliente(self, cliente):
        self.cliente_list.append(cliente)

    def visualizar_clientes(self):
        return '\n'.join(map(str, self.cliente_list))

    def atualizar_cliente(self, index, cliente):
        if 0 <= index < len(self.cliente_list):
            self.cliente_list[index] = cliente
            print("Cliente atualizado com sucesso.")
        else:
            print("Índice inválido.")

    def excluir_cliente(self, index):
        if 0 <= index < len(self.cliente_list):
            del self.cliente_list[index]
            print("Cliente excluído com sucesso.")
        else:
            print("Índice inválido.")

    def get_cliente_list(self):
        return self.cliente_list

File: AtividadeJava/CadastroClientes.java
package AtividadeJava;

import java.io.Serializable;
import java.util.ArrayList;

public class CadastroClientes implements Serializable {

    private ArrayList<Cliente> clienteList;

    public CadastroClientes() {
        this.clienteList = new ArrayList<Cliente>();
    }

    public void adicionarCliente(Cliente cliente) {
        clienteList.add(cliente);
    }

    public String visualizarClientes() {
        StringBuilder sb = new StringBuilder();
        
        for (Cliente cliente : clienteList) {
            sb.append(cliente.toString()).append("\n");
        }
        
        return sb.toString();
    }

    public void atualizarCliente(int index, Cliente cliente) {
        if (index >= 0 && index < clienteList.size()) {
            clienteList.set(index, cliente);
            System.out.println("Cliente atualizado com sucesso.");
        } else {
            System.out.println("Índice inválido.");
        }
    }

    public void excluirCliente(int index) {
        if (index >= 0 && index < clienteList.size()) {
            clienteList.remove(index);
            System.out.println("Cliente excluído com sucesso.");
        } else {
            System.out.println("Índice inválido.");
        }
    }

    public ArrayList<Cliente> getClienteList() {
        return clienteList;
    }
}